clover/netcdf/describe.py

```diff
@@ -64,14 +64,18 @@ def describe(path_or_dataset):
         }
 
         if dtype not in ('str', ):
-            try:
+            if len(variable.shape) > 2:
+                # Avoid loading the entire array into memory by iterating along the first index (usually time)
+                variable_info.update({
+                    'min': min(variable[i, :].min().item() for i in range(variable.shape[0])),

```

Comment:
The only way we could do this without Python loops would be to use a [memmapped](https://docs.scipy.org/doc/numpy/reference/generated/numpy.memmap.html) array. But even there, we either have to load everything into memory and write it back to the memmaped array, or loop in Python to load chunks from the Dataset into the array on disk.

Otherwise, we either a) loop in Python, or b) write a Cython extension and loop there.

I also looked for ways to load the data flattened so that it would be easier to load predictable-sized chunks (dimensions aren't important here). But after much searching, I couldn't find any way to ravel the data before it was in memory.